Makefile

```diff
@@ -0,0 +1,8 @@
+all:
+	go build -o shim
```

Comment:
Can we name the binary `kata-shim` (to match `main.go:shimName`)? A binary called `shim` is too generic imho. This will require an update to `.gitignore` too.

I've raised a PR to do the same for the agent: https://github.com/kata-containers/agent/pull/39.